extensions/poetry/main.ts

```diff
@@ -50,6 +50,49 @@ async function findRoot(manifest: string): Promise<string | undefined> {
   return undefined;
 }
 
+// List with all of poetry's subcommands.
+const knownSubcommands = [
+  "about",
+  "add",
+  "build",
+  "check",
+  "config",
+  "export",
+  "help",
+  "init",
+  "install",
+  "list",
+  "lock",
+  "new",
+  "publish",
+  "remove",
+  "run",
+  "search",
+  "shell",
+  "show",
+  "update",
+  "version",
+  "cache",
+  "debug",
+  "env",
+  "self",
+  "source",
+];
```

Comment:
Do we have a way of future-proofing this? I'm not saying to add any code to the extension, per se, but I feel like actively monitoring new releases for every tool we support in a manual way could be prone to error. We could add a few scripts _somewhere_ that, say, check the `--help` output or something similar for exhaustiveness.

Definitely a matter for a separate issue and low priority, though.

Reply:
I've contemplated how we could future-proof this but I genuinely have no idea.

I hope this will serve just as a somewhat temporary solution until we can find something more proper. But as these things usually go, temporary solutions are the most permanent things there are.